android/app/src/main/java/org/openbot/env/NearbyConnection.java

```diff
@@ -165,21 +191,21 @@ private void stopDiscovery() {
     connectionsClient.stopDiscovery();
   }
 
-  /** Finds an opponent to play the game with using Nearby Connections. */
-  public void connect(Context context, PayloadCallback payloadCallback) {
-    NearbyConnection.context = context;
-    NearbyConnection.payloadCallback = payloadCallback;
+  @Override
+  public void connect(Context context) {
+    this.context = context;
     connectionsClient = Nearby.getConnectionsClient(context);
 
     // make sure we are not connecting
-    disconnect();
+    disconnect(context);
```

Comment:
Previously this did not require passing the context. Why is the context needed now?